docs/build/workflow-snapshots.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Workflow Snapshots
+sidebar_label: Workflow Snapshots
+slug: /workflow-snapshots
+---
+
+Workflow snapshots capture and store the state of a workflow and job code at a specific point in time based on a save operation and/or run. Snapshots are helpful with debugging, auditing, and improving overall workflow performance.
```

Comment:
@ritazagoni what is the "state of a workflow" vs. "job code"? 
Also I'm worried "state" will be confused with job "state"... what if we changed this language to something like the below (assuming the below is accurate)? 

```
Workflow snapshots capture and store the state/version of a workflow's configuration and job code at a specific point in time, when a workflow was updated or run.
```